### PR TITLE
depotdownloader: init at 2.4.1

### PR DESCRIPTION
**Pull request by babbaj**
###### Motivation for this change
A cli tool for downloading game files from steam servers

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125833` at 78d905eb run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>depotdownloader</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/depotdownloader/default.nix:37:5:
```
   |
37 |     installPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/tools/misc/depotdownloader/default.nix:25:5:
```
   |
25 |     buildPhase = ''
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/depotdownloader/deps.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
+fetchNuGet:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/depotdownloader/deps.nix:1]:
```suggestion
fetchNuGet:
```

File: pkgs/tools/misc/depotdownloader/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{ stdenv, lib, fetchFromGitHub, fetchurl, linkFarmFromDrvs, makeWrapper
+,  dotnet-sdk_5, dotnetPackages
+}:
+
+let
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/depotdownloader/default.nix:5]:
```suggestion
let
```

File: pkgs/tools/misc/depotdownloader/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+       name = "nuget-${name}-${version}.nupkg";
+        url = "https://www.nuget.org/api/v2/package/${name}/${version}";
+        inherit sha256;
+    };
+    deps = import ./deps.nix fetchNuGet;
+in
+stdenv.mkDerivation rec {
+    pname = "depotdownloader";
+    version = "2.4.1";
+    src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/depotdownloader/default.nix:16]:
```suggestion
    version = "2.4.1";

    src = fetchFromGitHub {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please indent with 2 spaces.

**Pull request merged**